### DISCUSSION TITLE
Provide example of using real typescript instead of default babel-typescript

### DISCUSSION THREAD

**Discussion by airtonix** [vercel/next.js#24735] [Ideas]
### Describe the feature you'd like to request

I'd like to be using real typescript and all of its features like parameter decorators so I can embrace aspect orientated programming with less boilerplate.

examples of projects that I can't use because babel-typescript is only simulated typescript:

- https://github.com/yggdrasilts/axiosfit
- https://www.npmjs.com/package/kaop-ts

Serverside I can use nestjs and all its aop approaches, but I miss that paradigm in my react frontend.

### Describe the solution you'd like

Some kind of example `next.config.js` that shows how to use real typescript instead of `babel-typescript`

### Describe alternatives you've considered

- quit developing
- throw the project away
- rewrite my entire project
- come here for ideas and help from the community

**Comment by QuentinRoy** [comment]
You may need to provide a bit more details about the specific problem you are encountering with babel-typescript. I am not a typescript expert, but my understanding is that there are no runtime differences between typescript code compiled with tsc, and  typescript code compiled with babel-typescript: https://www.typescriptlang.org/docs/handbook/babel-with-typescript.html#babel-vs-tsc-for-typescript.

**Comment by QuentinRoy** [comment]
It is possible your problem is that by default decorators are not supported (as they should be because their JS API is not yet finalized). If you want to enable them, it is possible with `@babel/plugin-proposal-decorators`. You can learn more here: https://babeljs.io/docs/en/babel-plugin-transform-typescript, and here: https://github.com/k1r0s/kaop-ts/blob/master/docs/faq.md#babel.

**Reply by airtonix** [reply]
Sorry for the late reply.

Unfortunately only real typescript will work.

If it were an option to use babel then the nestjs team would be suggesting that.

They don't.

So here we are.
